sklearn/utils/_param_validation.py

```diff
@@ -499,6 +499,30 @@ def __str__(self):
         )
 
 
+class HasMethod(_Constraint):
+    """Constraint representing objects that expose a specific method.
+
+    It is useful for parameters following a protocol and where we don't want to impose
+    an affiliation to a specific module or class.
+
+    Parameters
+    ----------
+    method : str
+        The method that the object is expected to expose.
+    """
+
+    @validate_params({"method": [str]})
+    def __init__(self, method):
+        super().__init__()
+        self.method = method
+
+    def is_satisfied_by(self, val):
+        return callable(getattr(val, self.method, None))
```

Comment:
I'm not sure if I'm qualified to drop a comment on your PR so pardon me!
Wouldn't it be better if the constraint checks for more than one method? Some estimators require implementing several methods, for example: https://github.com/scikit-learn/scikit-learn/blob/3c75d36c6585b175f29f207c20cced1222b4d1b7/sklearn/multiclass.py#L855-L856.